### ISSUE TITLE
Core/Creature: Creature Formations

### ISSUE THREAD

**Issue by CraftedRO** [TrinityCore/TrinityCore#29652]
### Description

Not sure why [initially issue](https://github.com/TrinityCore/TrinityCore/issues/17563) was closed but is far from being fixed at least on 3.3.5.

If you kill the leader of the formation, formation group members will stuck in place.

https://github.com/TrinityCore/TrinityCore/assets/24683355/287ff76b-354f-48f7-90ed-ea4e31231ead

### Expected behaviour

If you kill formation leader, most likely the formation group members shoud continue to follow the leader assigned path.

### Steps to reproduce the problem

.go c 43201 (or any other formation leader)
.kill (it ^)
notice

### Branch

3.3.5

### TC rev. hash/commit

b4783d6cace49ae0cfe2b0d5059ba776840b307b

### Operating system

Windows 11 x64

### Custom changes

[Passive anticheat](https://github.com/TrinityCore/TrinityCoreCustomChanges/commits/3.3.5-passive_anticheat)

**Comment by malcrom** [comment]
That system needs a full rewrite.

**Comment by Legoso** [comment]
As mentioned in issue #29653 , you killed Twilight Marauder Morna (NPC 15541) using the GM command `.die` and thereby avoided aggroing the 2 escort NPCs via their formation leader and its SAI and aggro trigger. If you had aggroed the formation leader, all 3 would have dismounted to attack your character.

**Comment by CraftedRO** [comment]
@Legoso you missunderstand this issue mate, it's not about how you aggro the formation, it's about what happen after the formation leader dies (the other alive members remain stuck in place without possibility to continue leader assigned path untill probably leader respawn and get back in formation).

**Comment by Legoso** [comment]
Fair enough. Malcrom's comment should have given me a hint that this is not about how the formation leader is stopped, only about how the remaining formation members are currently moving and how they should move if it had been coded correctly. I see your point. That should remain an open issue.
